mono/metadata/marshal.c

```diff
@@ -11345,11 +11340,11 @@ mono_marshal_get_thunk_invoke_wrapper (MonoMethod *method)
 			mono_mb_emit_op (mb, CEE_BOX, mono_class_from_mono_type (sig->ret));
 	}
 
-#ifdef USE_COOP_GC
-	/* XXX merge reset_blocking_end with detach */
-	mono_mb_emit_ldloc (mb, coop_gc_var);
-	mono_mb_emit_icall (mb, mono_threads_reset_blocking_end);
-#endif
+	if (mono_threads_is_coop_enabled ()) {
+		/* XXX merge reset_blocking_end with detach */
+		mono_mb_emit_ldloc (mb, coop_gc_var);
+		mono_mb_emit_icall (mb, mono_threads_reset_blocking_end);
```

Comment:
You can update master, I will rebase on it, a d resolve conflicts.

On Fri, 23 Oct 2015, 19:47 Aleksey Kliger (λgeek) <notifications@github.com>
wrote:

> In mono/metadata/marshal.c
> <https://github.com/mono/mono/pull/2169#discussion_r42900376>:
>
> > @@ -11345,11 +11340,11 @@ mono_marshal_get_thunk_invoke_wrapper (MonoMethod *method)
> >  			mono_mb_emit_op (mb, CEE_BOX, mono_class_from_mono_type (sig->ret));
> >  	}
> >
> > -#ifdef USE_COOP_GC
> > -	/* XXX merge reset_blocking_end with detach */
> > -	mono_mb_emit_ldloc (mb, coop_gc_var);
> > -	mono_mb_emit_icall (mb, mono_threads_reset_blocking_end);
> > -#endif
> > +	if (mono_threads_is_coop_enabled ()) {
> > +		/* XXX merge reset_blocking_end with detach */
> > +		mono_mb_emit_ldloc (mb, coop_gc_var);
> > +		mono_mb_emit_icall (mb, mono_threads_reset_blocking_end);
>
> And here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/mono/pull/2169/files#r42900376>.
>